control-plane/subcommand/gateway-resources/command.go

```diff
@@ -288,7 +290,15 @@ func (c *Command) Run(args []string) int {
 	}
 
 	if len(c.gatewayConfig.GatewayClassConfigs) > 0 {
-		err = c.createV2GatewayClassAndClassConfigs(context.Background(), "consul-mesh-gateway", "consul-mesh-gateway-controller")
+		err = c.createV2GatewayClassAndClassConfigs(context.Background(), meshGatewayComponent, "consul-mesh-gateway-controller")
+		if err != nil {
+			c.UI.Error(err.Error())
+			return 1
+		}
+	}
+
+	if len(c.gatewayConfig.MeshGateways) > 0 {
```

Comment:
I thought about a check here that an entry in the gatewayClassConfig list must exist for each mesh gateway and panic if not (as this would be a programmer error on our end), but left it out for now

Reply:
Can you clarify what you mean? Are you saying you thought about:
```
if len(c.gatewayConfig.MeshGateways) > 0 {
  if cantFindMatch(c.gatewayConfig.MeshGateways, c.gatewayConfig.GatewayClassConfigs){
     error("cannot find matching gatewayClassConfig for `my-mesh-gw`, `my-mesh-gw-cc` does not exist")
    }
....
```
I don't know how we would want a user to try and fix that, maybe we need to catch it before it gets here? idk like bats tests or something. I feel like if this is failing then someone messed with the gatewayResources configMap and shouldn't have.
I don't have any suggestions here, just talking it out.

Reply:
Yeah definitely something for us to keep in mind.